src/win_permissions.rs

```diff
@@ -0,0 +1,268 @@
+use winapi::um::winnt::*;
+use winapi::um::accctrl::*;
+use winapi::um::aclapi::*;
+use winapi::um::securitybaseapi::*;
+use winapi::um::minwinbase::{LPTR, SECURITY_ATTRIBUTES, PSECURITY_ATTRIBUTES};
+use winapi::um::winbase::{LocalAlloc, LocalFree};
+use winapi::shared::winerror::ERROR_SUCCESS;
+
+use std::ptr;
+use std::io;
+use std::mem;
+use std::marker;
+
+pub struct SecurityAttributes {
+    attributes: Option<InnerAttributes>,
+}
+
+impl SecurityAttributes {
+    pub fn empty() -> SecurityAttributes {
+        SecurityAttributes { attributes: None }
+    }
+
+    pub fn allow_everyone_connect() -> io::Result<SecurityAttributes> {
+        let attributes = Some(InnerAttributes::allow_everyone(GENERIC_READ | FILE_WRITE_DATA)?);
+        Ok(SecurityAttributes { attributes })
+    }
+
+    pub fn allow_everyone_create() -> io::Result<SecurityAttributes> {
+        let attributes = Some(InnerAttributes::allow_everyone(GENERIC_READ | GENERIC_WRITE)?);

```

Comment:
`GENERIC_WRITE` allows anyone to create another listening pipe, which probably isn't what most people want. However, Node.js clients will always require `GENERIC_WRITE` on read and write pipes for reasons that are beyond me - https://github.com/nodejs/node/blob/4e7d82945dfc3f9e264c258ef29c45910b7fd8bc/deps/uv/src/win/pipe.c#L123.